zipkin-web/src/main/scala/com/twitter/zipkin/web/Handlers.scala

```diff
@@ -315,7 +315,7 @@ class Handlers(mustacheGenerator: ZipkinMustache, queryExtractor: QueryExtractor
         "parentId" -> span.parentId.filter(spanMap.get(_).isDefined).map(SpanId(_).toString),
         "spanName" -> span.name,
         "serviceNames" -> span.serviceNames.mkString(","),
-        "serviceName" -> span.serviceName,
+        "serviceName" -> span.serviceName.getOrElse(""),
```

Comment:
I am not sure we do have a contract to never return spans without service name. We said we can't *find* such spans by service name (obviously), but if they are retrieved as part of a larger trace, I think we need to return them. They may contain lots of interesting information useful even without a service name.